lib/modules/ipfs/upload.js

```diff
@@ -12,48 +12,51 @@ class IPFS {
   }
 
   deploy() {
-    console.log("deploying!");
-    let self = this;
-    async.waterfall([
-      function findBinary(callback) {
-        let ipfs_bin = shelljs.which(self.configIpfsBin);
-
-        if (ipfs_bin === 'ipfs not found' || !ipfs_bin) {
-          console.log(('=== WARNING: ' + self.configIpfsBin + ' not found or not in the path. Guessing ~/go/bin/ipfs for path').yellow);
-          ipfs_bin = "~/go/bin/ipfs";
+    return new Promise((resolve, reject) => {
+      console.log("deploying!");
+      let self = this;
+      async.waterfall([
+        function findBinary(callback) {
+          let ipfs_bin = shelljs.which(self.configIpfsBin);
+
+          if (ipfs_bin === 'ipfs not found' || !ipfs_bin) {
+            console.log(('=== WARNING: ' + self.configIpfsBin + ' not found or not in the path. Guessing ~/go/bin/ipfs for path').yellow);
+            ipfs_bin = "~/go/bin/ipfs";
+          }
+
+          return callback(null, ipfs_bin);
```

Comment:
No need to return

Reply:
Just saw it's not directly your code, but could you please clean the returns on the last callbacks of each function :)